scripts/simplify.py

```diff
@@ -0,0 +1,215 @@
+	for trace in traces:
+		new_trace = xml.createElement('path')
+
+		# Recreate each allowed attribute in the newly created element
+		for allowed_attr in trace_allowed_attrs:
+			# Check if field has a value
+			if not trace.getAttribute(allowed_attr):
+				continue
+		
+			new_trace.setAttribute(allowed_attr, trace.getAttribute(allowed_attr))
+
+		# Assign default trace style
+		new_trace.setAttribute('style', default_trace_style)
+
+		# Add UUID if needed
+		if not new_trace.getAttribute('pcbre:uuid'):
+			new_trace.setAttribute('pcbre:uuid', str(uuid.uuid4()))
+
+		# Sort attributes
+		sort_element_attr(new_trace)
+
+		new_layer.appendChild(new_trace)
+
+	# Sort elements
+	new_layer.childNodes.sort(key=lambda x: x.getAttribute('pcbre:uuid'))
+
+def main(argv):
+    # Load and parse SVG
+	input_xml = minidom.parse(argv[0])
+	input_svg = input_xml.getElementsByTagName('svg')[0]
+
+	# Get 'layers' from SVG
+	layers = []
+
+	for element in input_xml.getElementsByTagName('g'):
+		# Sanity check
+		if element.getAttribute('inkscape:groupmode') != 'layer':
+			raise Exception('None layer groups are not supported at this time!')
+
+		if not element.getAttribute('pcbre:uuid'):
+			print('WARNING: No UUID assigned to layer. Skipping layer', element.getAttribute('inkscape:label'))
+			continue
+
+		layers += [element]
+
+	# Sort layers
+	layers.sort(key=lambda x: x.getAttribute('pcbre:uuid'))
+
+	# Create our 'clean' SVG object
+	clean_xml = minidom.Document()
+	clean_svg = clean_xml.createElement('svg')
+
+	# Recreate each allowed attribute
+	svg_allowed_attrs = ['width', 'height', 'viewBox']
+	for svg_allowed_attr in svg_allowed_attrs:
+		# Check if field has a value
+		if not input_svg.getAttribute(svg_allowed_attr):
+			continue
+	
+		clean_svg.setAttribute(svg_allowed_attr, input_svg.getAttribute(svg_allowed_attr))
+
+	# Add back XML namespcaes
+	input_svg_attributes = list(input_svg.attributes.keys()) if input_svg.attributes else []
+	for attribute in input_svg_attributes:
+		if attribute.startswith('xmlns:'):
+			clean_svg.setAttribute(attribute, input_svg.getAttribute(attribute))
+
+	# Sort attributes
+	sort_element_attr(clean_svg)
+
+	# Process each layer
+	layer_allowed_attrs = ['inkscape:label', 'transform', 'pcbre:uuid']
+	for layer in layers:
+		print('Processsing layer:', layer.getAttribute('inkscape:label'))
+
+		# Create group element
+		new_layer = clean_xml.createElement('g')
+
+		# Set default attributes
+		for layer_attr, value in default_layer_attrs.items():
+			new_layer.setAttribute(layer_attr, value)
+
+		# Recreate each allowed attribute in the newly created element
+		for svg_allowed_attr in layer_allowed_attrs:
+			# Check if field has a value
+			if not layer.getAttribute(svg_allowed_attr):
+				continue
+		
+			new_layer.setAttribute(svg_allowed_attr, layer.getAttribute(svg_allowed_attr))
+
+		# Add UUID if needed
+		if not element.getAttribute('pcbre:uuid'):
+			element.setAttribute('pcbre:uuid', str(uuid.uuid4()))
```

Comment:
I'm worried about some tools eating custom tags (I'm not sure how/if SVG enables extensions and if all tools support it properly?) and would have abused the ID tag with a suffix (also because editors will display it then, so we can find it in the file if we generate a warning).

Reply:
Using the id attribute appears to be the correct way to implement this functionality.
https://www.w3.org/TR/SVG2/struct.html#Core.attrib

Tested with Inkscape and it appears to work correctly.

Fixed as a side effect of f3277365787f9aa8112f77eaeca83fb2549c81c3 though not finalized as it's more directly addressed in another comment.